Sources/WalletConnectSign/Services/HistoryService.swift

```diff
@@ -39,4 +67,25 @@ private extension HistoryService {
             expiry: request.request.expiry
         )
     }
+    
+    func mapProposeParams(_ record: RPCHistory.Record) -> SessionType.ProposeParams? {
+        guard let proposal = try? record.request.params?.get(SessionType.ProposeParams.self)
+        else { return nil }
+        return proposal
+    }
+    
+    func mapProposalRecord(_ record: RPCHistory.Record) -> Session.Proposal? {
+        guard let proposal = try? record.request.params?.get(SessionType.ProposeParams.self)
+        else { return nil }
+        
+        return Session.Proposal(
```

Comment:
you can use SessionProposal's `publicRepresentation(pairingTopic: String) -> Session.Proposal` here